src/hooks/useBasePopoverReactionList/index.ts

```diff
@@ -0,0 +1,131 @@
+import {useEffect, useRef, useState} from 'react';
+import type {SyntheticEvent} from 'react';
+import {Dimensions} from 'react-native';
+import * as EmojiUtils from '@libs/EmojiUtils';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import type {BasePopoverReactionListHookProps, ReactionListAnchor, ShowReactionList} from './types';
+
+export default function useBasePopoverReactionList({emojiName, emojiReactions, accountID, reportActionID, preferredLocale}: BasePopoverReactionListHookProps) {
```

Comment:
any reason we extracted this to a hook? why can't we have the function inside the component?

Reply:
I did it because of a better code readability, maintainability, and reusability. If we wanted to include all these logic and dirty works inside the component itself, we would've ended up with a messy code base (same as the class component that was before), that nobody could read and understand in a glance. Custome hooks usually make development experience much better if used correctly. Not to mention that the logic becomes way more simplified, and the component itself doesn't end up with many lines as it was before.


Here are some references for this:

https://dev.to/rohanfaiyazkhan/react-custom-hooks-what-why-and-how-5053
https://www.benmvp.com/blog/how-to-extract-react-component-logic-custom-hook/
https://dev.to/rasaf_ibrahim/a-guide-to-react-custom-hooks-2b4h

Also in the `STYLE.md` file, it's mentioned, and I quote:

> Class components are DEPRECATED. Use function components and React hooks.

I believe @blazejkustra already reviewed this [here](https://github.com/Expensify/App/pull/34549#discussion_r1464791677). Please let me know if you have another opinion based on this.
Thanks.

Reply:
Looks like we will not use any of the functions outside `BasePopoverReactionList`, i am not sure if this really is a good practice. cc @blazejkustra

Reply:
Thanks @getusha 
Please let me know if any further changes are required.